dev/Pager/PrototypePager/PrototypePager.cs

```diff
@@ -0,0 +1,100 @@
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+
+// The Templated Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234235
+
+namespace MUXControlsTestApp
+{
+    public sealed partial class PrototypePager : Control
+    {
+        public enum PagerDisplayModes { Auto, ComboBox, NumberBox, NumberPanel, }
+        public enum ButtonVisibilityMode { Auto, AlwaysVisible, HiddenOnEdge, None, }
+
+        private Button FirstPageButton, PreviousPageButton, NextPageButton, LastPageButton;
+
+        public event TypedEventHandler<PrototypePager, PageChangedEventArgs> PageChanged;
+        
+        public PrototypePager()
+        {
+            this.DefaultStyleKey = typeof(PrototypePager);
+            this.Loaded += (s, args) => { PagerTemplateSettings = new PagerTemplateSettings(this); };
+
+        }
+        protected override void OnApplyTemplate()
+        {
+            // Grab UIElements for later
+            FirstPageButton = GetTemplateChild<Button>("FirstPageButton");
+            PreviousPageButton = GetTemplateChild<Button>("PreviousPageButton");
+            NextPageButton = GetTemplateChild<Button>("NextPageButton");
+            LastPageButton = GetTemplateChild<Button>("LastPageButton");
+            FirstPageButtonTestHook = FirstPageButton;
+            PreviousPageButtonTestHook = PreviousPageButton;
+            NextPageButtonTestHook = NextPageButton;
+            LastPageButtonTestHook = LastPageButton;
+            NumberBoxDisplayTestHook = GetTemplateChild<NumberBox>("NumberBoxDisplay");
+            ComboBoxDisplayTestHook = GetTemplateChild<ComboBox>("ComboBoxDisplay");
+
+            // Attach Callbacks for property changes
+            RegisterPropertyChangedCallback(SelectedIndexProperty, DisablePageButtonsOnEdge);
+            RegisterPropertyChangedCallback(SelectedIndexProperty, (s, e) => { PageChanged?.Invoke(this, new PageChangedEventArgs(SelectedIndex - 1)); });
+
+            // Attach click events
+            FirstPageButton.Click += (s, e) => { SelectedIndex = 1; };
+            PreviousPageButton.Click += (s, e) => { SelectedIndex -= 1; };
+            NextPageButton.Click += (s, e) => { SelectedIndex += 1; };
+            LastPageButton.Click += (s, e) => { SelectedIndex = NumberOfPages; };
+
+            InitializeDisplayMode();
+        }
+
+        T GetTemplateChild<T>(string name) where T : DependencyObject
+        {
+            T templateChild = GetTemplateChild(name) as T;
+            if (templateChild == null)
+            {
+                throw new NullReferenceException(name);
+            }
+            return templateChild;
+        }
+
+        private void InitializeDisplayMode()
+        {
+            switch (PagerDisplayMode)
+            {
+                case PagerDisplayModes.NumberBox:
+                    VisualStateManager.GoToState(this, "NumberBoxVisible", false);
```

Comment:
>NumberBoxVisible [](start = 56, length = 16)

This string is shared across a number of your partial classes, can we pull it out into a constant?

Reply:
Made into a constant